### PR TITLE
[REG-1665] Remove all the old bot creation documentation

### PR DESCRIPTION
**Pull request by RG-nAmKcAz**
- Removes old documentation for RGAction/RGState , botruntime, etc
- Removes javascript docs (out of date / not currently supported)

image

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

File: docs/creating-bots/csharp/csharp-unity-bots.mdx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+```csharp
+/**
+ * Outline of this Bot algorithm. For every update:
+ *  - If enough time has passed for the bot to process a new update, handle the update.
+ *  - Select a random nearby player or enemy if one exists
+ *    - Follow that entity at distance 2f
+ *    - Perform a skill targeting that entity and position
+ */
+public class RunTurnBot: MonoBehaviour, IRGBot
```
Comment (abeizer) [inline docs/creating-bots/csharp/csharp-unity-bots.mdx:24]: `RunTurn` is awkward to see now since we've removed that terminology from the "early days". Want to make this something super general like "MyBot"? I think that would also better imply that the user can name the class whatever they want

**Review by abeizer — APPROVED** [review]
Couple of non-blocking comments:

File: sidebars.js [hunk 2]
```diff
@@ -50,22 +40,10 @@ const sidebars = {
           type: 'category',
           label: 'C# Reference',
```
[the remark this answers is not in the document]
Comment (RG-nAmKcAz) [reply sidebars.js]: updated

File: docs/creating-bots/csharp/csharp-unity-bots.mdx (hunk 1 above)
Comment (RG-nAmKcAz) [reply docs/creating-bots/csharp/csharp-unity-bots.mdx]: updated

File: package.json [hunk 3]
```diff
@@ -21,7 +21,8 @@
     "clsx": "^1.2.1",
     "prism-react-renderer": "^1.3.5",
     "react": "^17.0.2",
-    "react-dom": "^17.0.2"
+    "react-dom": "^17.0.2",
+    "yarn": "^1.22.22"
```
Comment (vontell) [inline package.json:25]: (non-blocking, not really a big deal) Hmmm not sure if this should be here, typically yarn is installed globally for most folks.

**Review by vontell — APPROVED** [review]

**Comment by RG-nAmKcAz** [comment]
@vontell & @abeizer See the newer commits.  I have taken a pass to update agent builder, tutorial, and at least parts of the github workflow, as well as addressing broken links in various places.  It is not perfect (isn't intended to be) but should get us back in the land of 'mostly correct' for our doc site

**Comment by RG-nAmKcAz** [comment]
I stopped short of recreating all new screenshots/etc so there are still a few small places where JS code shows up

**Review by vontell — APPROVED** [review]

**Review by abeizer — APPROVED** [review]
